src/server/main.py

```diff
@@ -0,0 +1,370 @@
+                self._tot_time = 0
+
+            self._override = bool(request.form["override"])
+
+            self._snapshot = JsonWriter(
+                str(utils.PATH_TO_LOGS),
+                self._name,
+                request.form["bench_id"],
+                metho
+            )
+
+            return {"busy": False}
+
+        return None
+
+    def route_status(self):
+        """Provide a way to check the status of the server."""
+        return {"status": "free"} if self._runner is None else {"status": "busy"}
+
+    def route_snapshot(self):
+        """
+        Return the latest computed snapshot of the current solving.
+
+        If no solving is running a placeholder is returned with generation set to -1.
+
+        No expected input
+
+        Returns :
+
+        {
+            "has_next":"bool",
+            "snapshot":"[Individual]",
+            "generation":"int"
+        }
+
+        The snapshot type is defined in the json_io class.
+
+        """
+        if self._runner is None:
+            return {
+                "has_next": False,
+                "snapshot": [],
+                "generation": -1
+            }
+
+        self._count += 1
+        compute = time.thread_time()
+        gen = next(self._runner)
+
+        while self._count != self._nb_it and self._count % self._rate != 0:
+            gen = next(self._runner)
+            self._count += 1
+
+        compute = time.thread_time() - compute
+        self._tot_time += compute
+
+        base = {
+                "has_next": not self._count == self._nb_it,
+                "snapshot": [],
+                "generation": self._count
+            }
+
+        base["snapshot"] = self._snapshot.add_snapshot(gen, self._tot_time)["individuals"]
+
+        if self._count == self._nb_it:
+            if self._override or self._name not in utils.get_logs():
+                self._snapshot.dump()
+            self._runner = None
+
+        return base
+
+    def route_benchmarks(self):
+        """Provide a list of all the available benchmarks."""
+        return utils.get_datasets()
+
+    def route_benchmark(self, bench_id: str):
+        """
+        Load and return a specific benchmark.
+
+        The benchmark is returned in the same format as provider by the benchmark reader.
+        """
+        bench = utils.parse_dataset(bench_id)
+        bench["STATIONS"] = list(bench["STATIONS"])
+        return bench
+
+    def route_logs(self):
+        """
+        Provide a list of all available logs.
+
+        The snapshot provided is the last used.
+
+        returns :
+
+        [
+            {
+                bench_id: str,
+                method: ga|drl
+                snapshots: "snaphshot"
+            }
+        ]
+
+        """
+        logs = []
+
+        for log in utils.get_logs():
+            logs.append(get_header(utils.PATH_TO_LOGS, log))
+
+        return logs
+
+    def route_results(self):
```

Comment:
No, it will contains all solutions that were snapshoted. This is done like this to allow to reproduce the same graphs as if the solutions were live computed. A sample of the exact data stored can be found in the test folder.

Reply:
All the results of any resolution can be viewed at the logs page.  The result page shows the final solution (the optimal so). I don’t have time to re-change all this right now.

Reply:
I think there is a misunderstanding about the pages and the routes. The results route send the whole log file. The logs route return the list of logs headers. A log header contains various information, namely the last snapshot produced. I do not think there is anything to change to either the front or the back here 😁. If needed I can rename this route to have a more explicit name.

Reply:
After a private discussion, a solution has been found and implemented.